common/utils/ti_fec.h

```diff
@@ -0,0 +1,108 @@
+#pragma once
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include "board_config_common.h"
+
+#define DEBUG_PRINTF(a...) \
+  {}
+// #define DEBUG_PRINTF(...) printf(__VA_ARGS__);
+
+#define TI_FEC_CRC_LEN_BYTES 2
+
+class FecEncoder {
+ public:
+  explicit FecEncoder(bool appendCRC = false) : m_appendCRC(appendCRC) {}
+
+  /*
+   * Encode a message. CRC will be appended, then FEC applied, then whitening
+   * happens. The result can be retrieved via FecEncoder::OutputArray
+   */
+  void Encode(uint8_t* input, size_t inLen);
+
+  inline const size_t OutputSize(const size_t inLen) {
+    if (m_appendCRC)
+      return 4 * ((inLen + TI_FEC_CRC_LEN_BYTES) / 2 + 1);
+    else
+      return 4 * ((inLen) / 2 + 1);
+  }
+  inline uint8_t* OutputArray() { return interleaved; }
+
+ private:
+  bool m_appendCRC;
+
+  //! input buffer + Trellis Terminator + CRC
+  //! I probably only need one of these arrays, but an extra 300 bytes shouldn't
+  //! be a huge deal?
+  uint8_t input[MAX_PACKET_SIZE + TI_FEC_CRC_LEN_BYTES + 2];
+  uint8_t fec[4 * ((MAX_PACKET_SIZE + TI_FEC_CRC_LEN_BYTES) / 2 + 1)];
+  uint8_t interleaved[4 * ((MAX_PACKET_SIZE + TI_FEC_CRC_LEN_BYTES) / 2 + 1)];
+};
+
+class FecDecoder {
+ public:
+  FecDecoder() {
+    Reset();
+    FillLUT();
+  }
+
+  ~FecDecoder() = default;
+
+  void Reset();
+
+  /**
+   * Decode an input buffer and store the result in the outputBuffer.
+   * The input buffer should be of size ((payload length+crc)/2+1)*4, and the
+   * input of size (payload length+crc)
+   * @param pInputMessage where to find the FEC encoded message
+   * @param pOutputBuffer where to store the decoded message
+   * @param decodedMessageLen How long we expect the decoded message to be.
+   */
+  void FecDecode(uint8_t* pInputMessage, uint8_t* pOutputBuffer,
+                 size_t decodedMessageLen);
+
+ private:
+  /**************************************************************************************************************
+   * @fn fecDecode
```

Comment:
Your docstrings don't use a consistent format. We don't really have a good standard elsewhere but probably worth keeping it consistent in the same file.